awx/main/tasks.py

```diff
@@ -3088,8 +3088,8 @@ def pod_definition(self):
             pod_spec_override = parse_yaml_or_json(self.task.instance.instance_group.pod_spec_override)
         pod_spec = {**default_pod_spec, **pod_spec_override}
 
-        pod_spec['spec']['containers'][0]['image'] = ee.image
-        pod_spec['spec']['containers'][0]['args'] = ['ansible-runner', 'worker', '--private-data-dir=/runner']
+        if pod_spec['spec']['containers'][0]['image'] is None or pod_spec['spec']['containers'][0]['image'] == "":
```

Comment:
This feels wrong to me. 

I think the right thing to do would change the `image:` field in the default pod spec to say something like:

 ```
image: <placeholder> # Value comes from the resolved Execution Environment at runtime
```

Reply:
If I override the pod spec's image, I would think my configuration would be taken into account. Changing the EE to actually change the image is confusing.